plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/handlers/FordiacQuickFixHandler.java

```diff
@@ -91,23 +99,76 @@ public static IMarker[] getMarkersFromEditorSelection(final IStructuredSelection
 				.map(FordiacQuickFixHandler::getMarker).toArray(IMarker[]::new);
 	}
 
+	public static boolean hasMarker(final EditPart editPart) {
+		if (FordiacAnnotationUtil
+				.getAnnotationModel(editPart) instanceof final FordiacMarkerGraphicalAnnotationModel annotationModel) {
+			final var annotation = annotationModel.getAnnotations(editPart.getModel());
+			return !annotation.isEmpty();
+		}
+		return false;
+	}
+
 	public static IMarker getMarker(final EditPart editPart) {
 		if (FordiacAnnotationUtil
 				.getAnnotationModel(editPart) instanceof final FordiacMarkerGraphicalAnnotationModel annotationModel) {
 			final var annotation = annotationModel.getAnnotations(editPart.getModel());
+
 			if (!annotation.isEmpty()
 					&& annotation.iterator().next() instanceof final GraphicalMarkerAnnotation markerAnnotation) {
```

Comment:
What if the first annotation is not a marker annotation? I would suggest using https://github.com/eclipse-4diac/4diac-ide/blob/c4b326463d87c6a02d05fdeeee0cbef7d0070bce/plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/handlers/FordiacQuickFixHandler.java#L123-L126 from below to find a marker or validation annotation to fix.